Lib/types.py

```diff
@@ -60,7 +60,27 @@ def _m(self): pass
 
 # Provide a PEP 3115 compliant mechanism for class creation
 def new_class(name, bases=(), kwds=None, exec_body=None):
-    """Create a class object dynamically using the appropriate metaclass."""
+    """Create a class object dynamically using the appropriate metaclass.
+
+    Arguments:
+    name -- the new class name
+    bases -- sequence of parent classes (default ())
+    kwds -- mapping of keyword arguments to be supplied to parents' `__init_subclass__`
+    exec_body -- a callback used to populate the freshly created class namespace
+
+    Note: the class-creation functionality of `new_class` is similar to `type`. However,
+    calls to `type` cannot be directly substituted by calls to `new_class`. For example:
```

Comment:
Although, you were hit but this particular problem, I don't think it deserves a dedicated note. Just describing clearly every argument is enough. Plus maybe an example of usage like this:
```
For example, the class definition::

    class C(A, B, foo=bar):
        x = 42

can be translated to the following call to `new_class`::

    C = new_class('C', (A, B), {'foo': bar}, lambda ns: ns.update(x=42))
```